pkg/proto/dispatch/v1/01_codec.go

```diff
@@ -62,6 +74,12 @@ func (vtprotoCodec) Unmarshal(data []byte, v any) error {
 	return fmt.Errorf("failed to unmarshal, message is %T, want proto.Message", v)
 }
 
-func init() {
-	encoding.RegisterCodec(vtprotoCodec{})
-}
+func init() { encoding.RegisterCodec(vtprotoCodec{}) }
+
+// Implement the preMarshaler interface
+
+func (r *DispatchCheckResponse) PreMarshaledBytes() []byte              { return r.Marshaled }
```

Comment:
Why not just call the field `premarshaled_bytes` and then have the interface be `func GetPremarshaledBytes()`? that way, you don't need to manually implement it

Reply:
This was a super good point. I totally forgot about the getters that get generated.